crates/sui-framework/sources/crypto/drand.move

```diff
@@ -0,0 +1,180 @@
+            Drand {
+                id: object::new(ctx),
+                previous_round: 2475343,
+                previous_sig: x"880c20aa83669a234f7b633917e43b7afb4e72421f0f520f7f570559356ed72e15867a77513346fbd37401709fc2192c1830936f23a751c228d091527290b304f2c6fa93851ca92c5ae7c3f7d7e0d06c65b561a6d6afeabce080f442f68d3845"
+            }
+        )
+    }
+
+    /// Advance `drand` by one round
+    public fun advance(drand: &mut Drand, sig: vector<u8>) {
+        let round = drand.previous_round + 1;
+        verify_drand_signature(sig, drand.previous_sig, round);
+        drand.previous_sig = sig;
+        drand.previous_round = round;
+    }
+
+    /// Advance `drand` by an arbitrary number of rounds.
+    /// Aborts if `round` is smaler than the most recent round `drand` has seen
+    public fun jump(drand: &mut Drand, sig: vector<u8>, previous_sig: vector<u8>, round: u64) {
+        assert!(round > drand.previous_round, ERoundAlreadyPassed);
+        verify_drand_signature(sig, previous_sig, round);
+        drand.previous_sig = sig;
+        drand.previous_round = round;
+    }
+
+    /// Return the most recent round we have seen
+    public fun round(drand: &Drand): u64 {
+        drand.previous_round
+    }
+
+    /// Return the most recent signature we have seen
+    public fun sig(drand: &Drand): vector<u8> {
+        drand.previous_sig
+    }
+
+    /// Return the most recent Unix time we have seen in seconds, according to drand
+    /// This time is coarse-grained--it will only advance by increments of PERIOD (30 seconds).
+    /// Note that this is not an up-to-date time unless `drand` has been updated recently
+    /// --it should only be used as a proof that this particular time occurred in the past.
+    /// Finally, note that although drand is supposed to produce a new round every 30 seconds
+    /// and always has since GENESIS, it is possible for the gaps between rounds to be > or < 30 seconds.
+    /// Thus, this function should only be used as an approximation of Unix time for non-production use-cases
+    /// --for anything security-critical, please use either Sui epoch time or the forthcoming https://github.com/MystenLabs/sui/pull/6490
+    public fun approx_unix_time_seconds(drand: &Drand): u64 {
+        GENESIS + (PERIOD * drand.previous_round)
+    }
+
+    /// Return `true` if `unix_timestamp` is confirmed to be in the past based on drand updates.
+    /// Note that a return value of `false` does *not* necesarily mean that `unix_timestamp` is in
+    /// in the future--`drand` might just be out-of-date.
+    public fun timestamp_has_passed(drand: &Drand, unix_timestamp: u64): bool {
+        unix_timestamp < approx_unix_time_seconds(drand)
+    }
+
+    // functions that leverage drand without requiring access to the shared object
+
+    /// Check that a given epoch time has passed by verifying a drand signature from a later time.
+    /// round must be at least (epoch_time - GENESIS) / PERIOD + 1).
```

Comment:
due to Genesis = round 1, we need to update this comment + logic here with `round - 2`